org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/resources/ProjectsProviderUtils.java

```diff
@@ -40,8 +41,24 @@ public static Collection<ISonarLintProject> allProjects() {
       .collect(Collectors.toSet());
   }
 
+  /**
+   *  Useful when we want SonarLint to behave differently when
+   *  - no project bound (ret = 0)
+   *  - all projects bound (ret = 1)
+   *  - at least one project bound (0 < ret < 1)
+   */
+  public static float boundToAllProjectsRatio() {
```

Comment:
Do you really make any use of this float? Why not return an enum:
```
enum WorkspaceProjectsBindingRatio {
    NONE_BOUND,
    SOME_BOUND,
    ALL_BOUND
}
```
It will make calling code more readable IMO.